src/app/pages/masterdata/department.functionality/department.functionality.component.ts

```diff
@@ -101,14 +113,64 @@ export class DepartmentFunctionalityComponent implements OnInit {
                         }
                     }
                 }
+                this.checkAllStatusView();
+                this.checkAllStatusOnlyHOD();
+            });
+    }
+     selectAllDeptView(value: any) : void{
+        if(value.checked)
+        {
+            this.pagesForDepartment.forEach(x =>{
```

Comment:
Instead of using true or false separately you can use "value.checked" directly in "x.AllowView" inside the forEach loop,
Please remove the If block